clusterloader2/pkg/measurement/common/ooms_tracker.go

```diff
@@ -230,7 +241,5 @@ func (m *clusterOOMsTrackerMeasurement) handleOOMEvent(_, obj interface{}) {
 		klog.Warningf(`unrecognized OOM event message pattern; event message contents: "%v"`, event.Message)
 	}
 
-	m.oomsLock.Lock()
```

Comment:
Same here, let's lock as narrowly as possible

Reply:
I disagree - there are few lines along each other, all needing a lock. Why would  you lock them separately?
It is not a heavy parallel scenario.

Reply:
Keep in mind that acquiring a lock has its cost too. There should be a balance between number of locks and amount of code guarded by it.

Reply:
I share Jacek's view on that - critical section ought to be as small as possible but in this particular case we need to have mutual exclusion applied to the whole piece of code (and, as stated, it shouldn't hurt us performance-wise).

If we were to do it more granularly (meaning we lock the read to `m.resourceVersionRecorded` and the write to `m.ooms` separately), we might end up with an undesired scenario in which two goroutines executing `handleOOMEvent` method that receive the same OOM event (with the same `ResourceVersion` that was previously unregistered) save that very OOM to `m.ooms` twice.

Reply:
I took a look at the Informer's code and there is only one go-routing for handling the events. It makes sense, otherwise a scenario like Antek described could happen. Given that, I agree with you that a single Lock/Unlock is better here